aesd-char-driver/aesd-circular-buffer.c

```diff
@@ -108,5 +122,33 @@ void aesd_circular_buffer_add_entry(struct aesd_circular_buffer *buffer, const s
 void aesd_circular_buffer_init(struct aesd_circular_buffer *buffer)
 {
     memset(buffer,0,sizeof(struct aesd_circular_buffer));
+    buffer->len=0;
+
+}
+
+void aesd_circular_buffer_deinit(struct aesd_circular_buffer *buffer)
+{
+	
+	int i;
+	struct aesd_buffer_entry *entry;
+
+	AESD_CIRCULAR_BUFFER_FOREACH(entry, buffer, i)
+	{
+
+		if(entry->buffptr != NULL)
+		{
+			
+#ifdef _KERNEL_
+			kfree(entry->buffptr);
```

Comment:
@chaisuresh rather than handle here I'd suggest handling in the application, since this will know how the buffer was allocated.  This is why the line below fails, the application isn't actually dynamically allocating the buffers in unit test.